satellite/audit/pathcollector_test.go

```diff
@@ -0,0 +1,112 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package audit_test
+
+import (
+	"context"
+	"math/rand"
+	"testing"
+	"time"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"storj.io/storj/internal/memory"
+	"storj.io/storj/internal/testcontext"
+	"storj.io/storj/internal/testplanet"
+	"storj.io/storj/internal/testrand"
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/storj"
+	"storj.io/storj/satellite/audit"
+	"storj.io/storj/storage"
+	"storj.io/storj/uplink"
+)
+
+// TestAuditPathCollector does the following:
+// - start testplanet with 5 nodes and a reservoir size of 3
+// - upload 5 files
+// - iterate over all the segments in satellite.Metainfo and store them in allPieces map
+// - create a audit observer and call metaloop.Join(auditObs)
+//
+// Then for every node in testplanet:
+//    - expect that there is a reservoir for that node on the audit observer
+//    - that the reservoir size is <= 2 (the maxReservoirSize)
+//    - that every item in the reservoir is unique
+//    - that looking up each pieceID in allPieces results in the same node ID
```

Comment:
Where do we do this check in the code?

Reply:
And why do we want to test for this?

Reply:
oops, we didn't end up checking it. we thought we wanted to check in order to make sure that each pieceID was associated with the correct storage node, but on second thought, I'm not sure we're able to do that. so I'm taking this commented line out and the `allPieces` map.

Reply:
Do we still nee that `satellite.Metainfo.Service.Iterate` block?

Reply:
no, I got rid of it